packages/Meshfree/test/tstNearestNeighborExodusGenerator.cpp

```diff
@@ -0,0 +1,259 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+
+#include "DTK_ConfigDefs.hpp"
+#include "DTK_Types.h"
+
+#include "PointCloudProblemGenerator/ExodusProblemGenerator.hpp"
+#include "PointCloudProblemGenerator/PointCloudProblemGenerator.hpp"
+#include <DTK_DetailsDistributedSearchTreeImpl.hpp>
+#include <DTK_NearestNeighborOperator.hpp>
+
+#include <Kokkos_View.hpp>
+
+#include <Teuchos_Array.hpp>
+#include <Teuchos_CommHelpers.hpp>
+#include <Teuchos_DefaultComm.hpp>
+#include <Teuchos_RCP.hpp>
+#include <Teuchos_UnitTestHarness.hpp>
+
+#include <cmath>
+#include <limits>
+#include <memory>
+#include <string>
+
+//---------------------------------------------------------------------------//
+// Brute-force calculate the nearest neighbors
+template <class Device>
+void computeNeighborsBruteForce(
+    const Teuchos::RCP<const Teuchos::Comm<int>> &comm,
+    const std::string &src_exodus_file, const std::string &tgt_exodus_file,
+    const Kokkos::View<DataTransferKit::Coordinate **, Kokkos::LayoutLeft,
+                       Device> &tgt_coords,
+    Kokkos::View<DataTransferKit::Coordinate **, Kokkos::LayoutLeft,
+                 Kokkos::Serial> &src_nearest_coords )
+{
+    // Make a new subcommunicator on each rank.
+    Teuchos::Array<int> subcomm_ranks( 1, comm->getRank() );
+    auto rank_subcomm = comm->createSubcommunicator( subcomm_ranks() );
+
+    // On each rank get the source coordinates with no partitioning and find
+    // the nearest neighbors to the locally-owned target points.
+    for ( int r = 0; r < comm->getSize(); ++r )
+    {
+        Kokkos::View<DataTransferKit::Coordinate **, Kokkos::LayoutLeft, Device>
+            src_coords;
+
+        // Load the source coordinates.
+        {
```

Comment:
Why do you have a curly brace around that bock?  Are you introducing a new scope on purpose to shadow the `tgt_coords` argument???  I was wondering how the call to `createUniquelyOwnedProblem()` even compiled since `tgt_coords` is a const reference in the declaration of `computeNeighborsBruteForce()`...

Reply:
yeah that is wrong shadowing is wrong - I have fixed this. The bracket scope is just so the dummy variables (who's names have been changed to reflect this) don't live after the loading of the exodus files.